etcdserver/etcdhttp/httptypes/errors.go

```diff
@@ -43,7 +43,10 @@ func (e HTTPError) WriteTo(w http.ResponseWriter) {
 	if err != nil {
 		plog.Panicf("marshal HTTPError should never fail (%v)", err)
 	}
-	w.Write(b)
+	if _, err := w.Write(b); err != nil {
+		plog.Errorf("http write errors with HTTPError (%v)", err)
```

Comment:
After thinking for a while, I think we probably make this function return an error.

I want to log like this `error writing HTTPError ("ABC") to 10.0.0.1:12345`. However, only the caller know the remote Addr.